imports/plugins/core/ui/client/components/media/mediaUploader.js

```diff
@@ -59,9 +59,43 @@ function MediaUploader(props) {
       });
     });
 
+
     Promise.all(promises)
-      .then(() => {
-        setIsUploading(false);
+      .then((responses) => {
+        // NOTE: This is a temporary workaround due to the fact that on the server,
+        // the sharp library generates product images in an async manner.
+        // A better solution would be to use subscriptions
+        const uploadedMediaIds = responses.map((response) => response.data.createMediaRecord.mediaRecord._id);
+
+        // Poll server every half a second to determine if all media has been successfully processed
+        let isAllMediaProcessed = false;
+        const timerId = setInterval(async () => {
+          const { data: { product } } = await refetchProduct();
+          const productMedia = [];
+          product.media.forEach((media) => {
+            const { id } = decodeOpaqueId(media._id);
+            productMedia.push({ id, thumbnailUrl: media.URLs.small });
+          });
+
+          isAllMediaProcessed = uploadedMediaIds.every((uploadedMediaId) => {
+            const mediaItem = productMedia.find((item) => item.id === uploadedMediaId);
+
+            // If a url has been generated, then these media items has been processed successfully.
+            return mediaItem && mediaItem.thumbnailUrl !== String(null);
+          });
+
+          if (isAllMediaProcessed) {
+            setIsUploading(false);
+            clearTimeout(timerId);
+          }
+        }, 500);
+
+        // Stop polling after 20 seconds
+        setTimeout(() => {
+          clearTimeout(timerId);
+          setIsUploading(false);
+        }, 20000);
+
```

Comment:
I'm seeing the progress bar sick around for the full 20 seconds on variants/options, even if the media shows up before then. Maybe something is off with the `isAllMediaProcessed` check above.

Reply:
fixed